lua/plenary/scandir.lua

```diff
@@ -0,0 +1,313 @@
+local Path = require'plenary.path'
+local os_sep = Path.path.sep
+
+local uv = vim.loop
+
+local m = {}
+
+local get_gitignore = function()
+  local data = vim.split(Path:new('.gitignore'):read(), '\n')
+  local gitignore = {}
+  for _, v in ipairs(data) do
```

Comment:
we might be able to stream `data` here? I don't remember if there is a `Path:iter` or anything, but that might be nice... or `Path:lines()` that returns an iterator. I wonder if we can get better speed / streamability with this method.

Reply:
There is not but i could write one later